src/engine/ChromaticFilter.cs

```diff
@@ -0,0 +1,34 @@
+using Godot;
+
+/// <summary>
+/// A chromatic aberration and barrel distrortion filter effect
+/// </summery>
+public class ChromaticFilter : TextureRect
+{
+    private static ChromaticFilter instance;
+    ShaderMaterial material;
+
+    public static ChromaticFilter Instance => instance;
+    public override void _Ready()
+    {
+        material = (ShaderMaterial) Material;
+        SetAmount(Settings.Instance.ChromaticAmount);
+        Show();
```

Comment:
Hiding this here if settings has this disabled would be fine for this PR.

Reply:
It now toggles based on user settings